File: codegen/results/v002/qwen3.6_35b_nothink/20260418_165425/typescript/005_unit_test_writer/solution.ts
function calculateDiscount(price: number, quantity: number): number {
    if (quantity < 10) return price * quantity;
    else if (quantity < 50) return price * quantity * 0.9;
    else return price * quantity * 0.75;
}

function runTest(description: string, price: number, quantity: number, expected: number): void {
    const result = calculateDiscount(price, quantity);
    if (Math.abs(result - expected) < 1e-9) {
        console.log(`PASS: ${description}`);
    } else {
        console.log(`FAIL: ${description} (expected: ${expected}, got: ${result})`);
    }
}

// Test 1: Below minimum threshold (quantity = 9) - no discount
runTest("No discount for quantity < 10 (quantity=9)", 10, 9, 90);

// Test 2: Boundary condition (quantity = 10) - 10% discount
runTest("10% discount for quantity >= 10 and < 50 (quantity=10)", 10, 10, 90);

// Test 3: Boundary condition (quantity = 49) - 10% discount
runTest("10% discount for quantity >= 10 and < 50 (quantity=49)", 10, 49, 441);

// Test 4: Boundary condition (quantity = 50) - 25% discount
runTest("25% discount for quantity >= 50 (quantity=50)", 10, 50, 375);

// Test 5: Above 50 threshold (quantity = 51) - 25% discount
runTest("25% discount for quantity >= 50 (quantity=51)", 10, 51, 382.5);

// Test 6: Zero quantity
runTest("Zero quantity returns 0", 10, 0, 0);

File: codegen/results/v002/qwen3.6_35b_nothink/20260418_165425/go/005_unit_test_writer/solution.go
package main

import (
	"fmt"
	"math"
)

func calculateDiscount(price float64, quantity int) float64 {
	if quantity < 10 {
		return price * float64(quantity)
	} else if quantity < 50 {
		return price * float64(quantity) * 0.9
	}
	return price * float64(quantity) * 0.75
}

func main() {
	testCases := []struct {
		description string
		price       float64
		quantity    int
		expected    float64
	}{
		{"Bulk discount for quantity 9 (less than 10)", 100.0, 9, 900.0},
		{"Bulk discount for quantity 10 (first threshold)", 100.0, 10, 900.0},
		{"Bulk discount for quantity 49 (just before second threshold)", 100.0, 49, 4410.0},
		{"Bulk discount for quantity 50 (second threshold)", 100.0, 50, 3750.0},
	}

	allPassed := true
	for _, tc := range testCases {
		result := calculateDiscount(tc.price, tc.quantity)
		if math.Abs(result-tc.expected) < 1e-9 {
			fmt.Printf("PASS: %s\n", tc.description)
		} else {
			fmt.Printf("FAIL: %s (expected: %.4f, got: %.4f)\n", tc.description, tc.expected, result)
			allPassed = false
		}
	}

	if allPassed {
		fmt.Println("\nAll tests passed!")
	} else {
		fmt.Println("\nSome tests failed.")
	}
}